lib/src/r_tree/non_leaf_node.dart

```diff
@@ -108,7 +107,7 @@ class NonLeafNode<E> extends Node<E> {
   }
 
   _convertToLeafNode() {
-    var nonLeafParent = parent as NonLeafNode<E>;
+    var nonLeafParent = parent as NonLeafNode<E>?;
```

Comment:
not introduced by your changes, but it seems like parent should just always be typed as NonLeafNode?